online_meetings/canvas_zoom_meetings.py

```diff
@@ -45,64 +38,96 @@
-        r = self.zoom_session.get(zoom_previous_url, params=data)
+        r = self.zoom_session.get(zoom_previous_url, params=kwargs)
         # Load in the json and look for results
         zoom_json = json.loads(r.text)
         if zoom_json and "result" in zoom_json:
             return zoom_json["result"]
         return None
 
+    @staticmethod
+    def extract_from_js(key: str, text: str) -> Optional[str]:
+        """Takes a javascript text and attempts to extract a key/value (May not work with everything)
+        
+        :param key: key to extract
+        :type key: str
+        :param text: text to search in
+        :type text: str
+        :return: The matching string or None if not found
+        :rtype: Optional[str]
+        """
+        pattern = re.search(f'{key}.*"(.*)"', text)
+        if pattern:
+            return pattern.group(1)
+        return None
+
     def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
         # Start up the zoom session
         # Initiate the LTI launch to Zoom in a session
         r = self.zoom_session.post(url=url, data=data)
+
+        # Get the scid
+        scid = self.extract_from_js("scid", r.text)
+        token = self.extract_from_js("X-XSRF-TOKEN", r.text)
+
         # Get the XSRF Token
-        pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
-        if pattern:
+        if token and scid:
             self.zoom_session.headers.update({
-                'X-XSRF-TOKEN': pattern.group(1)
+                'X-XSRF-TOKEN': token
             })
 
-            zoom_json = self.get_zoom_json(1)
+            zoom_json = self.get_zoom_json(page=1, lti_scid=scid)
             # The first call to zoom returns total and pageSize, get the total pages by dividing
             if zoom_json:
                 total = zoom_json["total"]
                 total_pages = math.ceil(total / zoom_json["pageSize"]) + 1
-            for page_num in range(1, total_pages):
+            for page in range(1, total_pages):
                 # Just skip the first call we've already called it, but still need to process
-                if page_num != 1:
-                    zoom_json = self.get_zoom_json(page_num)
-
-                for meeting in zoom_json["list"]:
-                    self.zoom_courses_meetings.append({
-                        'course_id': course_id,
-                        'meeting_id': meeting['meetingId'],
-                        'meeting_number': meeting['meetingNumber'],
-                        'host_id': meeting['hostId'],
-                        'topic': meeting['topic'],
-                        'join_url': meeting['joinUrl'],
-                        'start_time': meeting['startTime'],
-                        'status': meeting['status'],
-                        'timezone': meeting['timezone']
-                    })
+                if page != 1:
```

Comment:
Might have been some miscommunication here on my part. I was referring to his condition being clunky (in other words, that it would be clearer to process this first request outside the loop. Again, not a super big deal.

Reply:
Yeah, these Zoom API's are special since you need to do the first call to get the page size. It may be 0 and if it is we don't need to do anything but if it's > 0 we have to run this loop. I just figured to avoid the duplicate call to the API. Maybe this could be refactored but I think I'll leave it for the moment.